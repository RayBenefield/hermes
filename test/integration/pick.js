import flame from '@leonardvandriel/flame';
import * as databases from './database'; // eslint-disable-line import/no-unresolved,import/extensions,no-unused-vars
import uuid from '../utils/mock-uuid';
import random from '../utils/mock-random';
import configureSave from '../utils/save-db';
import configureDomain from '../../src/domain';
import configureFlame from '../../src/lib/local-db';

const db = configureFlame(flame);
const domain = configureDomain({ db, uuid, random });
const save = configureSave(flame); // eslint-disable-line no-unused-vars
beforeEach(() => flame.loadDatabase(undefined));

test('pick a card for the goal', () => {
    // Given
    const lead = {
        platform: 'facebook',
        id: 123456,
    };
    flame.loadDatabase(databases['shown-goal']);

    // When
    return domain({
        lead,
        action: 'pick',
        payload: {
            pick: 0,
            game: 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa',
            round: 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa',
        },
    })
        .then(save('picked-card'))
        .then(results => expect({
            results,
            db: flame.get('/'),
        }).toMatchSnapshot());
});

test('try to pick a card again', () => {
    // Given
    const lead = {
        platform: 'facebook',
        id: 123456,
    };
    flame.loadDatabase(databases['picked-card']);

    // When
    return domain({
        lead,
        action: 'pick',
        payload: {
            pick: 0,
            game: 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa',
            round: 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa',
        },
    })
        .then(results => expect({
            results,
            db: flame.get('/'),
        }).toMatchSnapshot());
});

test('the last player to pick a card', () => {
    // Given
    const lead = {
        platform: 'facebook',
        id: 123456,
    };
    flame.loadDatabase(databases['4-picked-cards']);

    // When
    return domain({
        lead,
        action: 'pick',
        payload: {
            pick: 0,
            game: 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa',
            round: 'aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa',
        },
    })
        .then(results => expect({
            results,
            db: flame.get('/'),
        }).toMatchSnapshot());
});
