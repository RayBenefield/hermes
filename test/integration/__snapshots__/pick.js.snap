// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show a hand with pickable cards 1`] = `
Object {
  "db": Object {
    "games": Object {
      "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
        "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        "notified_players": Object {
          "123456": true,
          "234567": true,
          "345678": true,
          "456789": true,
          "567890": true,
        },
        "players": Object {
          "123456": true,
          "234567": true,
          "345678": true,
          "456789": true,
          "567890": true,
        },
        "rounds": Object {
          "--------------------": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        },
      },
    },
    "hands": Object {
      "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
        "123456": Object {
          "cards": Object {
            "519": true,
            "520": true,
            "521": true,
            "522": true,
            "524": true,
            "525": true,
            "526": true,
            "527": true,
            "528": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 123456,
        },
        "234567": Object {
          "cards": Object {
            "0": true,
            "259": true,
            "260": true,
            "261": true,
            "262": true,
            "263": true,
            "783": true,
            "784": true,
            "786": true,
            "787": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 234567,
        },
        "345678": Object {
          "cards": Object {
            "0": true,
            "130": true,
            "131": true,
            "392": true,
            "393": true,
            "653": true,
            "654": true,
            "655": true,
            "915": true,
            "917": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 345678,
        },
        "456789": Object {
          "cards": Object {
            "0": true,
            "196": true,
            "327": true,
            "458": true,
            "589": true,
            "720": true,
            "850": true,
            "851": true,
            "981": true,
            "982": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 456789,
        },
        "567890": Object {
          "cards": Object {
            "0": true,
            "1014": true,
            "163": true,
            "19": true,
            "294": true,
            "32": true,
            "687": true,
            "818": true,
            "949": true,
            "98": true,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "player": 567890,
        },
      },
    },
    "players": Object {
      "facebook": Object {
        "123456": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 123456,
          "platform": "facebook",
        },
        "234567": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 234567,
          "platform": "facebook",
        },
        "345678": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 345678,
          "platform": "facebook",
        },
        "456789": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 456789,
          "platform": "facebook",
        },
        "567890": Object {
          "first_name": "Ray",
          "games": Object {
            "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
          },
          "id": 567890,
          "platform": "facebook",
        },
      },
    },
    "queue": Object {},
    "rounds": Object {
      "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
        "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": Object {
          "candidates": Object {
            "123456": 0,
          },
          "card": Object {
            "contents": "It's a D&D retroclone with _ added.",
            "id": 111,
          },
          "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
          "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        },
      },
    },
  },
  "results": Object {
    "action": "pick",
    "candidates": Object {},
    "game": Object {
      "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
      "notified_players": Object {
        "123456": true,
        "234567": true,
        "345678": true,
        "456789": true,
        "567890": true,
      },
      "players": Object {
        "123456": true,
        "234567": true,
        "345678": true,
        "456789": true,
        "567890": true,
      },
      "rounds": Object {
        "--------------------": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
      },
    },
    "lead": Object {
      "id": 123456,
      "platform": "facebook",
    },
    "messages": Array [
      Object {
        "payload": Object {
          "card": Object {
            "contents": "Flying sex snakes.",
            "id": 0,
          },
        },
        "type": "card-selected-message",
      },
      Object {
        "type": "wait-for-votes-message",
      },
    ],
    "payload": Object {
      "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
      "pick": 0,
      "round": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
    },
    "pick": Object {
      "contents": "Flying sex snakes.",
      "id": 0,
    },
    "player": Object {
      "first_name": "Ray",
      "games": Object {
        "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa": true,
      },
      "id": 123456,
      "platform": "facebook",
    },
    "round": Object {
      "card": Object {
        "contents": "It's a D&D retroclone with _ added.",
        "id": 111,
      },
      "game": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
      "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
    },
  },
}
`;
